webapi/src/main/scala/org/knora/webapi/responders/admin/PermissionsResponder.scala

```diff
@@ -861,148 +861,54 @@ final case class PermissionsResponder(
-    val permissionResourceClassChangeTask: Task[PermissionGetResponseADM] =
+  ): Task[DefaultObjectAccessPermissionGetResponseADM] = {
+    val task: Task[DefaultObjectAccessPermissionGetResponseADM] =
       for {
-        // get permission
-        permission <- permissionGetADM(permissionIri.value, requestingUser)
-        response <- permission match {
-                      // Is permission an administrative permission?
-                      case ap: AdministrativePermissionADM =>
-                        // Yes.
-                        ZIO.fail(
-                          ForbiddenException(
-                            s"Permission ${ap.iri} is of type administrative permission. " +
-                              s"Only a default object access permission defined for a resource class can be updated.",
-                          ),
-                        )
-                      case doap: DefaultObjectAccessPermissionADM =>
-                        // No. It is a default object access permission.
-                        for {
-                          _ <- updatePermission(
-                                 permissionIri = doap.iri,
-                                 maybeResourceClass = Some(changePermissionResourceClass.forResourceClass),
-                               )
-                          updatedPermission <- verifyUpdateOfResourceClass
-                        } yield DefaultObjectAccessPermissionGetResponseADM(
-                          updatedPermission.asInstanceOf[DefaultObjectAccessPermissionADM],
-                        )
-                      case _ =>
-                        ZIO.fail(
-                          UpdateNotPerformedException(
-                            s"Permission ${permissionIri.value} was not updated. Please report this as a bug.",
-                          ),
-                        )
-                    }
-      } yield response
-
-    IriLocker.runWithIriLock(apiRequestID, permissionIri.value, permissionResourceClassChangeTask)
+        doap <-
+          doapService.findById(permissionIri).someOrFail(NotFoundException(s"DOAP ${permissionIri.value} not found."))
+        resourceClassIri <- iriConverter
+                              .asSmartIri(changeRequest.forResourceClass)
+                              .mapError(_.getMessage)
+                              .flatMap(smartIri => ZIO.fromEither(ResourceClassIri.from(smartIri)))
+                              .mapError(BadRequestException(_))
+        newForWhat = doap.forWhat match
+                       case Group(_)                       => ResourceClass(resourceClassIri.toInternal)
+                       case ResourceClass(_)               => ResourceClass(resourceClassIri.toInternal)
+                       case Property(p)                    => ResourceClassAndProperty(resourceClassIri.toInternal, p)
+                       case ResourceClassAndProperty(_, p) => ResourceClassAndProperty(resourceClassIri.toInternal, p)
```

Comment:
I believe I understand your intention, but I guess you are too smart here. I think this behaviour is unintuitive. 

For example there is a DOAP xyz defined for a property, and now I send 

```
PUT 
{
    "forResourceClass": "new-res-class"
}
```

then I end up with this DOAP:

```
{
    "forResourceClass": "new-res-class",
    "forProperty": "old-prop",
}
```

But I argue that's not intended. If I had intended to get that, I would have sent this request in the first place:

```
PUT
{
    "forResourceClass": "new-res-class",
    "forProperty": "old-prop",
}
```

To turn it the other way around: With this PR's implementation, how can I be sure to end up with a DOAP that is solely for a property, but for nothing else? To be sure, I would have to delete the old DOAP and create a new one. 

So I argue to make a "hard overwrite", i.e. delete all the old stuff and add the new stuff.

This behaviour should then also be documented at https://docs.dasch.swiss/latest/DSP-API/03-endpoints/api-admin/permissions/#updating-a-default-object-access-permissions-resource-class

Reply:
The problem is the API is poorly designed for updating the combination of resource class and property:

We have two endpoints which only accept a single property:

`PUT admin/permissions/<doap_permissionIri>/property`
body: `{ "forProperty" : ...}`

`PUT /admin/permissions/<doap_permissionIri>/resourceClass`
body: `{ "forResourceClass" : ...}`

The combination of both parameters in the body - like you suggested - is currently not supported by the existing endpoints. Even the documentation omits the combination of class and property.

Reply:
That would be a huge improvement! I don't want to burden too much work on you, but the current situation is really subpar, and could lead to confusion.

Reply:
or create a new endpoint `PUT /admin/permissions/doap/<doap_permissionIri>/forWhat` which would accept every valid combination of group, property and resource class.

So you do not have to choose a different endpoint every time you want to change a doap.

Reply:
> PUT /admin/permissions/doap/<doap_permissionIri>/forWhat

Would also be fine for me

Reply:
I believe the old implementation did not create a ResourcClassAndProperty and just replaced it with either the property or resourceclass ForWhat. I will update this PR to be backwards compatible.